pydantic/_internal/_typing_extra.py

```diff
@@ -433,3 +433,10 @@ def is_dataclass(_cls: type[Any]) -> TypeGuard[type[StandardDataclass]]:
 
 def origin_is_type_alias_type(origin: Any) -> TypeGuard[TypeAliasType]:
     return isinstance(origin, TypeAliasType)
+
+
+def is_generic_alias(type_: type[Any]) -> bool:
+    if sys.version_info >= (3, 10):
+        return isinstance(type_, (types.GenericAlias, typing._GenericAlias))  # type: ignore[attr-defined]
+    else:
+        return isinstance(type_, typing._GenericAlias)  # type: ignore
```

Comment:
```suggestion
if sys.version_info >= (3, 10):
    def is_generic_alias(type_: type[Any]) -> bool:
        return isinstance(type_, (types.GenericAlias, typing._GenericAlias))  # type: ignore[attr-defined] 
else:
    def is_generic_alias(type_: type[Any]) -> bool:
        return isinstance(type_, typing._GenericAlias)  # type: ignore
```
Super minor, but eliminates a runtime branch 🤷‍♂️